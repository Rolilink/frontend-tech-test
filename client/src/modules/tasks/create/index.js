import connectToState from './providers/CreateTask';
import CreateTaskComponent from './components/CreateTask';

export default connectToState(CreateTaskComponent);
