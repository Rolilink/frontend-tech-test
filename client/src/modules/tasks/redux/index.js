import * as actionTypes from './actionTypes';
import * as factories from './factories';
import * as selectors from './selectors';
import * as utils from './utils';

import { fetchTasks, createTask } from './actions';

import reducerModule from './reducers';

const reducer = {
  name: 'tasks', reducer: reducerModule,
};

export {
  fetchTasks,
  createTask,
  actionTypes,
  factories,
  selectors,
  reducer,
  utils,
};
