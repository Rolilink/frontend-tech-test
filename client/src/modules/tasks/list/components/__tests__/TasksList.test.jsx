import React from 'react';
import { shallow, mount } from 'enzyme';
import serializer from 'enzyme-to-json/serializer';

import TasksList from '../TasksList';
import TaskListItem from '../TaskListItem';
import { taskFactory } from '../../../../../__tests__/testUtils/tasks';

expect.addSnapshotSerializer(serializer);

const tasks = [
  taskFactory({
    id: 1,
    title: 'Task Title',
    description: 'Task Description',
  }),
];

const tasksList = shallow(<TasksList tasks={tasks} />);
const getTaskListItem = tasksList.instance().getTaskListItem;

describe('Tasks.List.TasksList', () => {
  it('should render tasks', () => (
    expect(mount(<TasksList tasks={tasks} />)).toMatchSnapshot()
  ));

  it('#getTaskListItem should render an item', () => (
    expect(getTaskListItem(tasks[0])).toMatchSnapshot()
  ));

  it('#getTaskListItem should render an TaskListItem', () => (
    expect(getTaskListItem(tasks[0]).type).toEqual(TaskListItem)
  ));
});
