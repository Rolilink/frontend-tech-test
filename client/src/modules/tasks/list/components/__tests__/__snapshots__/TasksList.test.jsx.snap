// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tasks.List.TasksList #getTaskListItem should render an item 1`] = `
<TaskListItem
  className=""
  task={
    Object {
      "description": "Task Description",
      "id": 1,
      "title": "Task Title",
    }
  }
/>
`;

exports[`Tasks.List.TasksList should render tasks 1`] = `
<TasksList
  className=""
  tasks={
    Array [
      Object {
        "description": "Task Description",
        "id": 1,
        "title": "Task Title",
      },
    ]
  }
>
  <List
    className="TasksList "
    items={
      Array [
        Object {
          "description": "Task Description",
          "id": 1,
          "title": "Task Title",
        },
      ]
    }
    renderListItem={[Function]}
  >
    <ListGroup
      bsClass="list-group"
      className="TasksList "
    >
      <div
        className="TasksList  list-group"
      >
        <TaskListItem
          className=""
          task={
            Object {
              "description": "Task Description",
              "id": 1,
              "title": "Task Title",
            }
          }
        >
          <ListGroupItem
            bsClass="list-group-item"
            className="TaskListItem TaskListItem-1 "
            listItem={false}
          >
            <span
              className="TaskListItem TaskListItem-1  list-group-item"
            >
              <Row
                bsClass="row"
                componentClass="div"
              >
                <div
                  className="row"
                >
                  <Col
                    bsClass="col"
                    className="TaskListItem-CheckBoxContainer"
                    componentClass="div"
                    md={2}
                    sm={2}
                  >
                    <div
                      className="TaskListItem-CheckBoxContainer col-md-2 col-sm-2"
                    >
                      <Checkbox
                        bsClass="checkbox"
                        disabled={false}
                        inline={false}
                        title=""
                      >
                        <div
                          className="checkbox"
                        >
                          <label
                            title=""
                          >
                            <input
                              disabled={false}
                              type="checkbox"
                            />
                          </label>
                        </div>
                      </Checkbox>
                    </div>
                  </Col>
                  <Col
                    bsClass="col"
                    className="TaskListItem-TitleAndDescription"
                    componentClass="div"
                    md={8}
                    sm={8}
                  >
                    <div
                      className="TaskListItem-TitleAndDescription col-md-8 col-sm-8"
                    >
                      <Row
                        bsClass="row"
                        className="TaskListItem-TitleContainer"
                        componentClass="div"
                      >
                        <div
                          className="TaskListItem-TitleContainer row"
                        >
                          <Title
                            className="TaskListItem-Title"
                          >
                            <h5
                              className="Title TaskListItem-Title"
                            >
                              Task Title
                            </h5>
                          </Title>
                        </div>
                      </Row>
                      <Row
                        bsClass="row"
                        className="TaskListItem-DescriptionContainer"
                        componentClass="div"
                      >
                        <div
                          className="TaskListItem-DescriptionContainer row"
                        >
                          <Description
                            className="TaskListItem-Description"
                          >
                            <p
                              className="Description TaskListItem-Description"
                            >
                              Task Description
                            </p>
                          </Description>
                        </div>
                      </Row>
                    </div>
                  </Col>
                  <Col
                    bsClass="col"
                    className="TaskListItem-CheckBoxContainer"
                    componentClass="div"
                    md={2}
                    sm={2}
                  >
                    <div
                      className="TaskListItem-CheckBoxContainer col-md-2 col-sm-2"
                    >
                      <Button
                        active={false}
                        block={false}
                        bsClass="btn"
                        bsStyle="warning"
                        disabled={false}
                      >
                        <button
                          className="btn btn-warning"
                          disabled={false}
                          type="button"
                        >
                          Edit
                        </button>
                      </Button>
                    </div>
                  </Col>
                </div>
              </Row>
            </span>
          </ListGroupItem>
        </TaskListItem>
      </div>
    </ListGroup>
  </List>
</TasksList>
`;
